### DISCUSSION TITLE
Detect form changes in Alpine before a Livewire request is made?

### DISCUSSION THREAD

**Discussion by stuartcusackie** [filamentphp/filament#11721] [General]
Is this possible? I want to detect a change to a form in Alpine before a render request is made to Livewire.

Example:
```
<form wire:submit.prevent="calculate">
   {{ $this->form }}
</form>

<div x-data="{ form: @entangle('somehowEntangleForm??') }">
  <span x-if="form.dirty">User has made input</span>
</div>
```

So if I had a non-live field this would still pick up the change, if you get my drift.

Thanks!

**Comment by ArtMin96** [comment]
Is it about Panel builder or Forms component?

**Reply by stuartcusackie** [reply]
This is the Forms component on a custom livewire component.

**Reply by ArtMin96** [reply]
You can check this: [Feature: Unsaved changes alert](https://github.com/filamentphp/filament/pull/10516/files#diff-862adc3cb0c3797a11414ed9f516bef759aabf781a10ef6690779c1598162848) and [https://github.com/filamentphp/filament/pull/10516/files#diff-63bf14bc4a290f7d4adcca108b61a82bb5d441f0f57334c5b3e8a01e4cdf53f9](https://github.com/filamentphp/filament/pull/10516/files#diff-63bf14bc4a290f7d4adcca108b61a82bb5d441f0f57334c5b3e8a01e4cdf53f9)

**Comment by stuartcusackie** [comment]
Thanks. Looks interesting. I'm not sure if it's useful for my case, as I'm not using it for an alert.

For the moment I am just wrapping my form in a custom alpine component:

```
/*  components/pendingForm.js */
export default () => ({
    pending: true,

    init() {

        ['input', 'change'].forEach((eventName) => {
            this.$refs.form.addEventListener(eventName, (event) => {
                this.pending = true;
            });
        });
         
        // Can dispatch this event from livewire component to reset pending status
        Livewire.on('pending-form-complete', (event) => {
            this.pending = false;
        })
    }
})

/*  livewire/custom-component.blade.php */
<div x-data="pendingForm">        
        <form x-ref="form" wire:submit.prevent="calculate" class="space-y-6 w-full">
            {{ $this->form }}
        </form>

        <div x-show="pending">Form pending update</div>
</div>
```
